src/groupeddataframe/grouping.jl

```diff
@@ -520,24 +520,27 @@ a vector or tuple thereof, and `f` must be a callable. If `cols` is a single col
 with a named tuple holding `SubArray` views into these columns.
 
 If the first argument is a vector, tuple or named tuple of such pairs, each pair is
-handled as described above. If a named tuple, field names are used to name
+handled as described above. If it is a named tuple then field names are used to name
 each generated column.
 
 If the first argument is a callable `f`, it is passed a [`SubDataFrame`](@ref) view for each group,
-and the returned `DataFrame` then consists of the returned rows plus the grouping columns.
-If the returned data frame contains columns with the same names as the grouping columns,
+and the returned `GroupedDataFrame` then consists of the returned rows plus the grouping columns.
+If the value returned  by `f` contains columns with the same names as the grouping columns,
 they are required to be equal.
 Note that this second form is much slower than the first one due to type instability.
 
-`f` can return a single value, a row or multiple rows. The type of the returned value
-determines the shape of the resulting data frame:
-- A single value gives a data frame with a single column and one row per group.
-- A named tuple of single values or a [`DataFrameRow`](@ref) gives a data frame with one column
+`f` can return a single value, a row or multiple rows.
+In all cases, the resulting `GroupedDataFrame` contains all the grouping columns in addition
+to those generated by the application of `f`.
+The type of the returned value determines the shape of the resulting groups in `GroupedDataFrame`.
+Additional columns in groups are created using the following rules:
```

Comment:
I"m not sure repeating "in groups" really helps.
```suggestion
Additional columns are created using the following rules:
```